platform/security/certificate/security-ocsp-checker/src/main/java/org/codice/ddf/security/ocsp/checker/OcspChecker.java

```diff
@@ -123,127 +144,254 @@ Certificate convertToBouncyCastleCert(X509Certificate cert) throws OcspCheckerEx
    */
   @VisibleForTesting
   OCSPReq generateOcspRequest(Certificate cert) throws OcspCheckerException {
     try {
+      X509CertificateHolder issuerCert = resolveIssuerCertificate(cert);
+
       JcaDigestCalculatorProviderBuilder digestCalculatorProviderBuilder =
           new JcaDigestCalculatorProviderBuilder();
       DigestCalculatorProvider digestCalculatorProvider = digestCalculatorProviderBuilder.build();
       DigestCalculator digestCalculator = digestCalculatorProvider.get(CertificateID.HASH_SHA1);
 
       CertificateID certId =
-          new CertificateID(
-              digestCalculator, new X509CertificateHolder(cert), cert.getSerialNumber().getValue());
+          new CertificateID(digestCalculator, issuerCert, cert.getSerialNumber().getValue());
 
       OCSPReqBuilder ocspReqGenerator = new OCSPReqBuilder();
       ocspReqGenerator.addRequest(certId);
       return ocspReqGenerator.build();
 
     } catch (OCSPException | OperatorCreationException e) {
       postErrorEvent(
-          "Unable to create a OCSP request. The certificate status could not be verified. "
+          "Unable to create an OCSP request. The certificate status could not be verified. "
               + e.getMessage());
       throw new OcspCheckerException();
     }
   }
 
   /**
-   * Sends the {@param ocspReq} request and returns the returned Response
+   * Returns an {@link X509CertificateHolder} containing the issuer of the passed in {@param cert}.
+   * Search is performed in the system truststore.
    *
-   * @param ocspReq - the OCSP request to send
-   * @return the response from the OCSP server
-   * @throws OcspCheckerException after posting an alert to the admin console, if any error occurs
+   * @param cert - the {@link Certificate} to get the issuer from.
+   * @return {@link X509CertificateHolder} containing the issuer of the passed in {@param cert}.
+   * @throws OcspCheckerException if the issuer cannot be resolved.
    */
-  @VisibleForTesting
-  Response sendOcspRequest(OCSPReq ocspReq) throws OcspCheckerException {
+  private X509CertificateHolder resolveIssuerCertificate(Certificate cert)
+      throws OcspCheckerException {
+    X500Name issuerName = cert.getIssuer();
+
+    String trustStorePath =
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) SecurityConstants::getTruststorePath);
+    String trustStorePass =
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) SecurityConstants::getTruststorePassword);
+
+    if (isBlank(trustStorePath) || isBlank(trustStorePass)) {
+      throw new OcspCheckerException("Problem retrieving truststore properties.");
+    }
+
+    KeyStore truststore;
+
+    try (InputStream truststoreInputStream = new FileInputStream(trustStorePath)) {
+      truststore = SecurityConstants.newTruststore();
+      truststore.load(truststoreInputStream, trustStorePass.toCharArray());
```

Comment:
❓ We should check with @stustison and see if this is worth logging to the security audit log.

Reply:
IMO there isn't much value in auditing here.

As for auditing in the case of an error when reading the truststore, the error will propagate up and make its way into the logs / frontend.

And as for auditing the truststore operations in general, we are only *reading* from the truststore (a lot less dangerous than writing 😱), and we aren't using any user parameters or anything when reading the truststore, leaving little room for attack.

But getting the 👌 from @stustison would be 👌.

Reply:
@stustison Confirmed we _should_ audit here 🎉.